File: PlayMinds-Back/src/main/java/com/coderhouse/pmb/Entitys/Card.java
package com.coderhouse.pmb.Entitys;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "card")
public class Card {

    @Id
    @Column(name = "id_card")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idCard;
    private String name;
    private String description;
    private boolean correct;

    @ManyToOne
    @JoinColumn(name = "FK_id_img", referencedColumnName = "id_img")
    private Image image;
}

File: ExamplesOBJ/models/Entitys/Card.ts
import { Image } from "./Image";

export class Card {
  idCard?: number;
  name: string;
  description: string;
  correct: boolean;
  image: Image;

  constructor(
    idCard: number,
    name: string,
    image: Image,
    description: string,
    correct: boolean
  ) {
    this.idCard = idCard;
    this.name = name;
    this.image = image;
    this.description = description;
    this.correct = correct;
  }
}
